tensorboard/plugins/debugger/debugger_server_lib.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Receives data from a TensorFlow debugger. Writes event summaries.
+
+This listener server writes debugging-related events into a logdir directory,
+from which a TensorBoard instance can read.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+import json
+import os
+import re
+import threading
+import time
+
+import tensorflow as tf
+from tensorflow.python.debug.lib import grpc_debug_server
+
+from tensorboard.plugins.debugger import constants
+# pylint: disable=line-too-long
+from tensorboard.plugins.debugger import events_writer_manager as events_writer_manager_lib
+# pylint: enable=line-too-long
+from tensorboard.plugins.debugger import numerics_alert
+
+
+class DebuggerDataStreamHandler(
+    grpc_debug_server.EventListenerBaseStreamHandler):
+  """Implementation of stream handler for debugger data.
+
+  Each instance of this class is created by a DebuggerDataServer upon a
+  gRPC stream established between the debugged Session::Run() invocation in
+  TensorFlow core runtime and the DebuggerDataServer instance.
+
+  Each instance of this class does the following:
+    1) receives a core metadata Event proto during its constructor call.
+    2) receives GraphDef Event proto(s) through its on_graph_event method.
+    3) receives tensor value Event proto(s) through its on_value_event method.
+  """
+
+  def __init__(self,  # pylint: disable=super-init-not-called
```

Comment:
The super class __init__ method currently raises a NotImplementedError. I'll fix it on the tensorflow side first. Added a TODO item for myself here.